cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -341,10 +342,20 @@ func (c *Client) RemoveMember(id uint64) ([]Member, error) {
 	return ret, nil
 }
 
-// AddMember notifies an existing etcd cluster that a new member is joining, and
+// AddMember adds a new member into the etcd cluster
+func (c *Client) AddMember(name string, peerAddrs string) ([]Member, error) {
+	return c.addMember(name, peerAddrs, false, false)
+}
+
+// AddMemberAsLeanerAndPromote adds a new learner member into the etcd cluster and promotes it to a voting member
+func (c *Client) AddMemberAsLeanerAndPromote(name string, peerAddrs string) ([]Member, error) {
+	return c.addMember(name, peerAddrs, true, true)
+}
+
+// addMember notifies an existing etcd cluster that a new member is joining, and
 // return the updated list of members. If the member has already been added to the
 // cluster, this will return the existing list of etcd members.
-func (c *Client) AddMember(name string, peerAddrs string) ([]Member, error) {
+func (c *Client) addMember(name string, peerAddrs string, isLearner bool, promote bool) ([]Member, error) {
```

Comment:
maybe in that case we should also remove the promote bool snd always promote learners

Reply:
Removed.